### PR TITLE
Delete resource through Java Client

### LINKED ISSUE #9793: Support resource delete command in Java client
# Description
We want to be able to send resource deletion commands from the Java client.

- Send the request to the gateway
- Handle the response
- Required parameter: key of the resource

Blocked by: #9790

### PR DESCRIPTION
**Pull request by remcowesterhoud**
## Description

Adds Java client support for deletion of resources.

## Related issues

closes #9793 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by koevskinikola — APPROVED** [review]
LGTM @remcowesterhoud :+1: 

I just have a clarifying question and a small hint.

File: clients/java/src/test/java/io/camunda/zeebe/client/util/RecordingGatewayService.java [hunk 1]
```diff
@@ -313,6 +317,13 @@ public void modifyProcessInstance(
     handle(request, responseObserver);
   }
 
+  @Override
+  public void deleteResource(
+      final DeleteResourceRequest request,
+      final StreamObserver<DeleteResourceResponse> responseObserver) {
+    handle(request, responseObserver);
+  }
+
```
[the remark this answers is not in the document]
Comment (remcowesterhoud) [reply clients/java/src/test/java/io/camunda/zeebe/client/util/RecordingGatewayService.java:326]: I am struggling to figure out why this code exists for some commands. Not all commands have this `onXRequest` method. The methods seem to add a request handler, which also happens in the constructor. The difference is creating a default instance, versus creating an instance manually.

Thing is, I don't see the `onXRequest` methods being called anywhere other than tests. So it seems we are not setting these handlers at all? Are they just for testing purposes?

As we currently don't need it (as far as I can see 😄) I wouldn't want to add it. We didn't do this for the other commands either. But I would like to understand what this is for.

[the remark this answers is not in the document]
Comment (remcowesterhoud) [reply clients/java/src/test/java/io/camunda/zeebe/client/util/RecordingGatewayService.java:326]: Ah never mind I just realised what it is. The `RecordingGatewayService` is specifically used for testing. It's also in the test sources, not the regular sources. It's basically a "mock" of the Zeebe Gateway, providing an implementation to run the tests.

This verifies my opinion that it's for testing purposes only and really not necessary here.

**Comment by remcowesterhoud** [comment]
bors merge

**Pull request merged**